File: 2st/Riyeon/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String, Integer> hm = new HashMap();
        for(String s : participant){
            if(hm.get(s) != null)
                hm.put(s, hm.get(s)+1);
            else
                hm.put(s, 1);
        }
        for(String s : completion){
            if(hm.get(s) != null)
                hm.put(s, hm.get(s)-1);
        }
        for(String key : hm.keySet())
            if(hm.get(key) != 0)
                answer = key;
        return answer;
    }
}

File: 2st/Riyeon/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    d = {}
    for i in participant:
        if i in d:
            d[i] += 1
        else:
            d[i] = 1
    for i in completion:
        d[i] -= 1
        if d[i] == 0:
            del d[i]
        
    for i in d.keys():
        answer = i
    return answer